mypyc/test-data/irbuild-lists.test

```diff
@@ -573,6 +573,228 @@ L0:
     r0 = CPySequence_Sort(a)
     return 1
 
+[case testListFromRange1]
+def fn() -> list[int]:
+    return list(range(3))
```

Comment:
`list(range(...))` seems to be by far the most common of these use cases, so having some irbuild tests for them seems reasonable.